File: Swift/subdomain-visit-count.swift
class Solution {
    func subdomainVisits(_ cpdomains: [String]) -> [String] {
        var res = [String: Int]()

        for domain in cpdomains {
            let splits = domain.split(separator: " ").map { String($0) }
            let count = Int(splits[0])!
            let domain = splits[1]
            let frags = domain.split(separator: ".").map { String($0) }
            var curr = [String]()
            for i in Array(0..<frags.count).reversed() {
                curr.append(frags[i])
                let subdomain = curr.reversed().joined(separator: ".")
                res[subdomain] = (res[subdomain] ?? 0) + count
            }
         }

        return res.map { "\($0.value) \($0.key)" }
    }
}

File: Python/subdomain-visit-count.py
import collections

class Solution:
    def subdomainVisits(self, cpdomains):
        """
        :type cpdomains: List[str]
        :rtype: List[str]
        """
        result = collections.defaultdict(int)
        for domain in cpdomains:
            count, domain = domain.split()
            count = int(count)
            frags = domain.split('.')
            curr = []
            for i in reversed(range(len(frags))):
                curr.append(frags[i])
                result[".".join(reversed(curr))] += count

        return ["{} {}".format(count, domain) for domain, count in result.items()]
